### ISSUE TITLE
FFI for iroh-net

### ISSUE THREAD

**Issue by rklaehn** [n0-computer/iroh#1976]
Can we make it easy for people from other languages to use iroh-net? Just open connections and streams?

The iroh-net API is unlikely to change that much, and I think it might be quite useful for developers from other languages.

**Comment by dignifiedquire** [comment]
streams across ffi boundaries are quite hard, especially if trying to make them performant. not against it in general , but it is going to be a lot of work for each language

**Comment by rklaehn** [comment]
> streams across ffi boundaries are quite hard, especially if trying to make them performant. not against it in general , but it is going to be a lot of work for each language

Yeah, I discussed this idea briefly with @ramfox. I thought that our ffi approach would already deal with streams, but if it does not this is a lot of work.

**Comment by dignifiedquire** [comment]
There is http://github.com/n0-computer/iroh-c-ffi now, and we will revisit FFI again after 1.0 so closing for now
